addons/fix-uploaded-svgs/userscript.js

```diff
@@ -1,4 +1,64 @@
 export default async function ({ addon, console }) {
+  function fixEditorSVG(element) {
+    const svg = element.cloneNode(true);
+
+    // Fixes the previous issue of SVGs imported from Affinity Designer
+    if (svg.height.baseVal.valueAsString === "100%" && svg.width.baseVal.valueAsString === "100%") {
+      svg.removeAttribute("height");
+      svg.removeAttribute("width");
+    }
+
+    if (addon.settings.get("removeUnsupported")) {
+      // "dominant-baseline" needs this iframe to shift accordingly
+      const iframe = document.createElement("iframe");
+      iframe.setAttribute("src", "about:blank");
+      document.body.append(iframe);
+      iframe.contentDocument.body.appendChild(svg);
```

Comment:
@StickmanRed Could you remind me why is this iframe needed? I see that it gets created and deleted, and there are no other references to it. Without it the addon continues to work normally it seems. So why?

Reply:
(We want to be careful because SVGs may cointain dangerous \<script> tags, so it's not reasonable to inject arbitrary data to an iframe without triple-checking what we're doing)

Reply:
> @StickmanRed Could you remind me why is this iframe needed? I see that it gets created and deleted, and there are no other references to it. Without it the addon continues to work normally it seems. So why?

I thought dominant-baseline would need it, but if it works fine, then I guess that wasn't the problem...

Reply:
If possible, check whether that works without declaring an iframe.

Reply:
Scratch does ignore the `dominant-baseline` attribute, but I don't know how to account for the displacement; I've looked for a way to get baseline information from the font, but to no avail...

The current code requires a window and the whole SVG to find the text offset, because it needs the text offset in context of the inherited properties/stylesheets. I guess a safer code would be to go through the DOM, keeping track of all the attribute values, and _then_ place the text element with its computed style in the iframe?

Reply:
By the way, would a sandboxed iframe prevent the danger of <script> tags?

Reply:
> By the way, would a sandboxed iframe prevent the danger of <script> tags?

Possibly.

Reply:
So in summary, the iframe is needed for the `getBoundingClientRect()` calls, right?

Reply:
I think we can use `document.createDocumentFragment()` here. I will update the code shortly. @StickmanRed Please test your code still works as intended after my changes.

Reply:
> So in summary, the iframe is needed for the `getBoundingClientRect()` calls, right?

@StickmanRed I'm still not sure if this is really the case... I don't think the methods available to the `svg` object vary depending on whether it's part of a document or not...

It would also be nice if you could attach an SVG that is fixed by this code so that this case can be more easily tested.

Reply:
> > So in summary, the iframe is needed for the `getBoundingClientRect()` calls, right?
> 
> @StickmanRed I'm still not sure if this is really the case... I don't think the methods available to the `svg` object vary depending on whether it's part of a document or not...
> 
> It would also be nice if you could attach an SVG that is fixed by this code so that this case can be more easily tested.

From `blocks2image`: 
![test1](https://github.com/user-attachments/assets/09b70b1f-fe35-49b4-b1c5-885da5db4122)
Removing the dominant-baseline code and importing this into Scratch will show a slight upward shift of the text, since their dominant-baseline property is set to "middle"

`getBoundingClientRect()` is a window method, so any detached SVGs won't have access to it...

Reply:
> `getBoundingClientRect()` is a window method

Is it, really?

MDN says `The Element.getBoundingClientRect() method returns a [DOMRect](https://developer.mozilla.org/en-US/docs/Web/API/DOMRect) object providing information about the size of an element and its position relative to the [viewport](https://developer.mozilla.org/en-US/docs/Glossary/Viewport).`

So I guess we need to emulate a viewport, I don't think we have anything better than an iframe.